src/excmds.ts

```diff
@@ -4360,6 +4360,48 @@ export async function echo(...str: string[]) {
     return str.join(" ")
 }
 
+/** helper function for js and jsb
+ *
+ * -p to take an extra argument located at the end of str[]
+ * -s to load js script of a source file from the config path
+ *
+ * @hidden
+ */
+async function js_helper(str: string[]) {
+    let doSource = false
+    let jsContent = null
+    /* tslint:disable:no-unused-declaration */
+    /* tslint:disable:no-dead-store */
+    let JS_ARG = null
+
+    while (true) {
+        if (str[0].startsWith("-p")) {
+            JS_ARG = str[str.length - 1]
+            str = str.slice(1, -1)
+        } else if (str[0].startsWith("-s")) {
+            doSource = true
+            str = str.slice(1)
+        } else {
+            break
+        }
+    }
+
+    if (doSource) {
+        const sourceFilename = str.join(" ")
+        if (sourceFilename.search(/(^|[\/\\])..[\/\\]/) >= 0) {
+            throw new Error('Source Filename cannot contains "/../"')
+            return
+        }
+        const sep = "/"
+        const rcPath = (await Native.getrcpath()).split(sep).slice(0, -1)
+        const jsPath = [...rcPath, sourceFilename].join(sep)
+        jsContent = (await Native.read(jsPath)).content
+    } else {
+        jsContent = str.join(" ")
+    }
+    return eval(jsContent)
+}
+
 /**
  * Lets you execute JavaScript in the page context. If you want to get the result back, use `composite js ... | fillcmdline`
```

Comment:
Please document all the arguments here (and mention attaching things to `tri.`or `window.`) :)